src/closure_decorator.rs

```diff
@@ -5,65 +5,51 @@ use swc_core::ast::*;
 use swc_core::common::util::take::Take;
 use swc_core::common::DUMMY_SP;
 use swc_core::plugin::proxies::PluginSourceMapProxy;
-use swc_core::utils::{prepend_stmts, private_ident, quote_ident};
+use swc_core::utils::{prepend_stmts, private_ident, quote_ident, quote_str};
 use swc_core::visit::*;
 
 use crate::class_like::ClassLike;
 use crate::js_util::{
-  ident_expr, prop_access_expr, ref_expr, require_expr, string_expr, this_expr,
+  prop_access_expr, ref_expr, require_expr, string_expr, this_expr, undefined_expr,
 };
 use crate::prepend::prepend;
 use crate::span::{concat_span, get_prop_name_span};
 use crate::virtual_machine::VirtualMachine;
 
 /**
- * Name of the `register` function that is injected into all compiled source files.
+ * Flags used to identify the Functionless AST association sequences.
  *
- * ```ts
- * function register_8269d1a8(func, ast) {
- *   func[Symbol.for("functionless:AST")] = ast;
- *   return func;
- * }
- * ```
- *
- * All Function Declarations, Expressions and Arrow Expressions are decorated with
- * the `register` function which attaches its AST as a property.
+ * These can be used by downstream programs to discover and manipulate the injects
+ * functionless assignment sequences.
  */
-const REGISTER_FUNCTION_NAME: &str = "register_8269d1a8";
+// (REGISTER,stash=func,stash[]=ast,stash)
+const REGISTER_FLAG: &str = "REGISTER_8269d1a8";
+// (REGISTER_REF,stash[]=ast)
+const REGISTER_REF_FLAG: &str = "REGISTER_REF_8269d1a8";
+// // (BIND, ...)
+const BIND_FLAG: &str = "BIND_8269d1a8";
+// // (PROXY, ...)
+const PROXY_FLAG: &str = "PROXY_8269d1a8";
+
+const GLOBAL_THIS_NAME: &str = "global_8269d1a8";
```

Comment:
I think we can get rid of the hashes now. The chance of this occurring is almost zero

Reply:
Which hashes? the salt? or you just mean in the flag/command names?

Reply:
The salt on the end. That's just a crc of the word functionless lol

Reply:
It can be removed from the flag/command names, I originally tried to put the string there without setting stash, but something strips the string off as a no-op.